From: Peter Zahradnik <peter.zahradnik@znik.sk>
Date: Sun, 15 Jan 2017 14:17:37 +0000
Subject: use as-needed and add latomic

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 38e052b..6bfb309 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,8 @@ ifdef DEBUG
 endif
 override CFLAGS += -D_FILE_OFFSET_BITS=64 -DVERSTRING=\"$(RELEASE)\" \
 	$(hash_CFLAGS) $(glib_CFLAGS) $(sqlite_CFLAGS) -rdynamic $(DEBUG_FLAGS)
-LIBRARY_FLAGS += $(hash_LIBS) $(glib_LIBS) $(sqlite_LIBS) -lm
+LIBRARY_FLAGS += -Wl,--as-needed -latomic -lm
+LIBRARY_FLAGS += $(hash_LIBS) $(glib_LIBS) $(sqlite_LIBS)
 
 # make C=1 to enable sparse
 ifdef C
